### PR TITLE
WiP - Replace SparkfunDMX with esp_dmx

### PR DESCRIPTION
**Pull request by netmindz**
Reports of poor output, so try swapping to the newer and more complete esp_dmx library

set either the RX or TX pin number for send/receive then do full reboot

### CODE REVIEW

**Comment by arneboe** [comment]
Hey,
do you need any help with this PR?
Currently I am using my own hacked (and very outdated version) of wled with a custom esp_dmx usermod but would like to migrate to moonModules.

**Comment by netmindz** [comment]
Hi @arneboe 

I wrote this while the code was familiar after writing the dmx INPUT support, but didn't have hardware setup at the time to test. 

If you could give it a test and report back that would be most useful

File: wled00/dmx.cpp [hunk 1]
```diff
@@ -7,13 +7,57 @@
...
+  if(dmxReceivePin > 0 && dmxTransmitPin < 1) {
+    USER_PRINTF("Listening for DMX on pin %u\n", dmxReceivePin);
+    dmx_set_pin(dmxPort, dmxTransmitPin, dmxReceivePin, dmxEnablePin);
+  }
+  else if(dmxTransmitPin > 0 && dmxReceivePin < 1) {
+    USER_PRINTF("Sending DMX on pin %u\n", dmxTransmitPin);
+    dmx_set_pin(dmxPort, dmxTransmitPin, dmxReceivePin, dmxEnablePin);
+  }
+  else if (dmxTransmitPin > 0 && dmxReceivePin > 0) {
+    USER_PRINTLN("DMX needs to be set for either send or receive - only set tx or rx pin");
```
Comment (arneboe) [inline wled00/dmx.cpp:31]: Why are you only allowing input or output?
For rdm to work both input and output need to be enabled at the same time.
esp_dmx uses the tx pin to answer to rdm requests.

**Comment by arneboe** [comment]
When `dmx_driver_install` is called, wled does no longer open a wifi hotspot.
I am not sure why this happens, yet.  As soon as I remove the call to `dmx_driver_install` wled opens a  hotspot and everything works fine.

Before I start digging any deeper, I will update to the latest version of esp_dmx and see if the bug is still present.

File: wled00/dmx.cpp (hunk 1 above)
Comment (netmindz) [reply wled00/dmx.cpp:31]: I'm not using RDM (yet) I have a different branch where I try and add support for that

**Comment by netmindz** [comment]
> When `dmx_driver_install` is called, wled does no longer open a wifi hotspot. I am not sure why this happens, yet. As soon as I remove the call to `dmx_driver_install` wled opens a hotspot and everything works fine.
> 
> Before I start digging any deeper, I will update to the latest version of esp_dmx and see if the bug is still present.

My first guess is that it's an interrupt clash rather than any bug with esp_dmx

**Pull request closed without merging**